### ISSUE TITLE
Counter in objective function

### ISSUE THREAD

**Issue by bkj** [fchollet/keras#1959]
I'm trying to implement a loss function with a temperature parameter that changes over time, eg

```
def my_loss(y_true, y_pred):
  a =  K.mean(K.abs(y_pred - y_true), axis=-1) * temp
  b =  K.mean(K.square(y_pred - y_true), axis=-1) * (1 - temp)
  return a + b
```

but I can't figure out how to get `temp` to change inside the function.  This does not work (presumably because non-Theano variables get fixed at compile time -- but I'm a Theano novice so that's as far as I can reason)

```
def temp():
  global temp_
  temp_ -= 0.1
  return temp_

def my_loss(y_true, y_pred):
  a =  K.mean(K.abs(y_pred - y_true), axis=-1) * temp()
  b =  K.mean(K.square(y_pred - y_true), axis=-1) * (1 - temp())
  return a + b
```

Any help would be much appreciated.

EDIT : Also, if this is the wrong way/place to try to set this kind of temperature parameter, I'm open to other suggestions.

**Comment by bkj** [comment]
Writing the post helped me get it. For anyone who is interested:

```
import numpy as np
from pprint import pprint
from keras.models import Sequential
from keras.layers.core import Dense
from keras import backend as K

from theano import shared, function
from theano.tensor.shared_randomstreams import RandomStreams

# Fake data
X = np.random.uniform(0, 1, (1000, 10))
y = np.random.choice((0, 1), 1000)

# Define loss
temp = shared(1.)
def my_loss(y_true, y_pred):
  a =  K.mean(K.abs(y_pred - y_true), axis=-1) * temp
  b =  K.mean(K.square(y_pred - y_true), axis=-1) * (1 - temp)
  return (a * 0) + (b*0) + temp

# Define model
model = Sequential()
model.add(Dense(10, input_dim=10, trainable=False))
model.add(Dense(1, activation='sigmoid', trainable=False))
model.compile(optimizer='rmsprop', loss=my_loss)

# Train while decreasing temperature
while temp.get_value() >= 0:
  fitist = model.fit(X, y, nb_epoch=1)
  temp.set_value((temp.get_value() - 0.1))
```
